src/instruments/src/EFB/Presets/Widgets/LightPresets.tsx

```diff
@@ -0,0 +1,177 @@
+        // loading of presets only allowed when aircraft is powered (also the case in the wasm)
+        if (isPowered) {
+            setLoadPresetVar(presetID);
+            toast.success(`Loading Preset: ${presetID}: ${presetName}`, {
+                autoClose: 250,
+                hideProgressBar: true,
+                closeButton: false,
+            });
+        } else {
+            toast.warning('Aircraft needs to be powered to load presets.', {
+                autoClose: 1000,
+                hideProgressBar: true,
+                closeButton: false,
+            });
+        }
+    };
+
+    // Sets the LVAR to tell the wasm to save the current lighting setting into the preset
+    const savePreset = (presetID: number) => {
+        // Saving of presets only allowed when aircraft is powered (also the case in the wasm)
+        if (isPowered) {
+            setSavePresetVar(presetID);
+            toast.success(`Saving Preset: ${presetID}: ${presetName}`, {
+                autoClose: 250,
+                hideProgressBar: true,
+                closeButton: false,
+            });
+        } else {
+            toast.warning('Aircraft needs to be powered to save presets.', {
+                autoClose: 1000,
+                hideProgressBar: true,
+                closeButton: false,
+            });
+        }
+    };
+
+    // User specified name for the current preset
+    const [presetName, setPresetName] = useState('');
+
+    // Sets the preset name locally and stores it into the parent persistent storage
+    const changePresetName = (newName: string): void => {
+        setPresetName(newName);
+        props.storePresetName(props.presetID, newName);
+    };
+
+    // Get preset name from persistent store when the names map changes
+    useEffect(() => {
+        const tmp: string = props.getPresetName(props.presetID);
+        setPresetName(tmp);
+    }, [props.namesMap]);
+
+    return (
+        <div className="flex flex-row justify-between my-1">
+            <div className="flex justify-center items-center w-24">
+                {props.presetID}
+            </div>
+            <div
+                className="flex justify-center items-center mx-4 w-full h-28 text-theme-text bg-theme-accent rounded-md border-2 border-theme-accent"
+            >
+                <SimpleInput
+                    className="w-80 text-2xl font-medium text-center"
+                    placeholder="No Name"
+                    value={presetName}
+                    onBlur={(value) => changePresetName(value)}
+                    maxLength={16}
+                />
+            </div>
+            <div
+                className="flex justify-center items-center mx-4 w-full h-28 text-theme-text hover:text-theme-body bg-theme-accent hover:bg-theme-highlight rounded-md border-2 border-theme-accent transition duration-100 items-centerh-24"
+                onClick={() => loadPreset(props.presetID)}
```

Comment:
Extract this into a handler function

Reply:
Aren't these already handler functions?
Did you mean renaming them?

Reply:
See above